avocado/core/job.py

```diff
@@ -462,6 +465,7 @@ def _run(self):
         """
         self._setup_job_results()
         self.__start_job_logging()
+        self.job_pre_post_dispatcher.map_methods('pre', self)
```

Comment:
Hi @ldoktor , indeed these two questions are worth thinking about.  And many more can actually come up.  So far we don't support a specific job pre or post state, that is, we don't say that the pre-job plugins will be given a job with x and y but not z.  IMHO, this is something we'll learn, and very quickly.  If this get's really popular, we may even end up having to add other "hook" positions, say a "pre create test suite" or "post discover tests" or whatever.

I think that we don't know what we don't know, and this first implementation can help us to learn just that.

Regarding `--dry-run`, IMHO it should behave like other job level "sub systems" such as sysinfo.  But I'm not even sure how `--dry-run` and sysinfo play along today...